### PR TITLE
Budget-app

### PR DESCRIPTION
**Pull request by roshan-bajgain**
### This branch includes the following features:
  - Home page
    - A simple page with the name of the app and links to the signup and log-in pages.
  - Sign up and log in to pages
    - The users can register in the app with their full name, email, and password.
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in.
 - Categories page
   - When the user logs in, they are presented with the categories page.
   - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
   - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
   - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
   - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- Added tests for models and views

### Presentation [Video](https://www.loom.com/share/f6e7c3c07bf4450085c8a76d0cfc592f)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Original design: [Gregoire Vella](https://www.behance.net/gregoirevella) on [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
+- Inspiration: Microverse
+
+## 📝 License
+
+This project is [MIT](./LICENSE.md) licensed.
```
Comment (Whoistolu) [inline README.md:112]: - [ ] [REQIRED] A very good project 👍🏽. However, @roshan-bajgain you need to ensure that this mobile web application is deployed using Heroku, Netlify, or GitHub Pages in order to make this project more accessible by any user and also make the project more professional.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @roshan-bajgain 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (roshan-bajgain) [reply README.md]: @Whoistolu, Thank you for your suggestion

**Review by Deepakdanger — APPROVED** [review]
Hi @roshan-bajgain, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**